xonsh/environ.py

```diff
@@ -120,6 +121,7 @@ def to_debug(x):
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
     'XONSH_HISTORY_SIZE': (is_history_tuple, to_history_tuple, history_tuple_to_str),
     'XONSH_LOGIN': (is_bool, to_bool, bool_to_str),
+    'XONSH_SHOW_TRACEBACK': (is_tb_opt, to_tb_tuple, tb_tuple_to_str),
```

Comment:
This feels a little strange to me (the fact that we are changing the form of the environment variable).  Can't we just store the repr of the option to detype, and use ast.literal_eval to retype?  That way, the option as the user set it stays intact.

I think the checks that are currently in `to_tb_tuple` could just as easily live directly in print_exception.